src/pool.rs

```diff
@@ -188,7 +191,25 @@ impl<Conn: Connection> PoolInner<Conn> {
                 // Handle requests from clients
                 request = self.rx.recv() => {
                     match request {
-                        Some(Request::Claim { tx }) => self.claim_or_enqueue(tx).await,
+                        Some(Request::Claim { tx }) => {
+                            self.claim_or_enqueue(tx).await
+                        }
+                        // The caller has explicitly asked us to terminate, and
+                        // we should respond to them once we've stopped doing
+                        // work.
+                        Some(Request::Terminate { tx }) => {
+                            // Terminate all background tasks, including:
+                            // - The resolver (may or may not have background
+                            // tasks, this is dependent on the implementation)
+                            // - Each of the slot sets
```

Comment:
take it or leave it: this comment already exists on the `terminate` method, perhaps we could just reference that so that it's not necessary to keep them in sync?